tests/phpunit/tests/html-api/wpHtmlProcessorBreadcrumbs.php

```diff
@@ -0,0 +1,370 @@
+			'ABBR',
+			'ACRONYM', // Neutralized
+			'ADDRESS',
+			'APPLET', // Deprecated
+			'AREA',
+			'ARTICLE',
+			'ASIDE',
+			'AUDIO',
+			'BASE',
+			'BDI',
+			'BDO',
+			'BGSOUND', // Deprecated; self-closing if self-closing flag provided, otherwise normal.
+			'BLINK', // Deprecated
+			'BODY',
+			'BR',
+			'BUTTON',
+			'CANVAS',
+			'CAPTION',
+			'CENTER', // Neutralized
+			'CITE',
+			'COL',
+			'COLGROUP',
+			'DATA',
+			'DATALIST',
+			'DD',
+			'DEL',
+			'DETAILS',
+			'DEFN',
+			'DIALOG',
+			'DL',
+			'DT',
+			'EMBED',
+			'FIELDSET',
+			'FOOTER',
+			'FORM',
+			'FRAME',
+			'FRAMESET',
+			'H1',
+			'H2',
+			'H3',
+			'H4',
+			'H5',
+			'H6',
+			'HEAD',
+			'HEADER',
+			'HGROUP',
+			'HR',
+			'HTML',
+			'IFRAME',
+			'INPUT',
+			'INS',
+			'ISINDEX', // Deprecated
+			'KBD',
+			'KEYGEN', // Deprecated; void
+			'LABEL',
+			'LEGEND',
+			'LI',
+			'LINK',
+			'LISTING', // Deprecated, use PRE instead.
+			'MAIN',
+			'MAP',
+			'MARK',
+			'MARQUEE', // Deprecated
+			'MATH',
+			'MENU',
+			'META',
+			'METER',
+			'MULTICOL', // Deprecated
+			'NAV',
+			'NEXTID', // Deprecated
+			'NOBR', // Neutralized
+			'NOEMBED', // Neutralized
+			'NOFRAMES', // Neutralized
+			'NOSCRIPT',
+			'OBJECT',
+			'OL',
+			'OPTGROUP',
+			'OPTION',
+			'OUTPUT',
+			'PICTURE',
+			'PLAINTEXT', // Neutralized
+			'PRE',
+			'PROGRESS',
+			'Q',
+			'RB', // Neutralized
+			'RP',
+			'RT',
+			'RTC', // Neutralized
+			'RUBY',
+			'SAMP',
+			'SCRIPT',
+			'SECTION',
+			'SELECT',
+			'SLOT',
+			'SOURCE',
+			'SPACER', // Deprecated
+			'SPAN',
+			'STYLE',
+			'SUB',
+			'SUMMARY',
+			'SUP',
+			'SVG',
+			'TABLE',
+			'TBODY',
+			'TD',
+			'TEMPLATE',
+			'TEXTAREA',
+			'TFOOT',
+			'TH',
+			'THEAD',
+			'TIME',
+			'TITLE',
+			'TR',
+			'TRACK',
+			'UL',
+			'VAR',
+			'VIDEO',
+			'WBR',
+			'XMP', // Deprecated, use PRE instead.
+
+			// Made up elements, custom elements.
+			'X-NOT-AN-HTML-ELEMENT',
+			'HUMAN-TIME',
+		);
+
+		$data = array();
+		foreach ( $unsupported_elements as $tag_name ) {
+			$data[ $tag_name ] = array( "<{$tag_name}>" );
+		}
+
+		return $data;
+	}
+
+	/**
+	 * @ticket 58517
+	 *
+	 * @dataProvider data_unsupported_markup
+	 *
+	 * @param string $html HTML containing unsupported markup.
+	 */
+	public function test_fails_when_encountering_unsupported_markup( $html, $description ) {
+		$p = WP_HTML_Processor::createFragment( $html );
+
+		while ( $p->step() && null === $p->get_attribute( 'supported' ) ) {
+			continue;
+		}
+
+		$this->assertTrue( $p->get_attribute( 'supported' ), 'Did not find required supported element.' );
+		$this->assertFalse( $p->step(), "Didn't properly reject unsupported markup: {$description}" );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[]
+	 */
+	public function data_unsupported_markup() {
+		return array(
+			'A with formatting following unclosed A' => array(
+				'<a><strong>Click <a supported><big unsupported>Here</big></a></strongu></a>',
```

Comment:
The `</strongu>` is a typo though, right?

```suggestion
				'<a><strong>Click <a supported><big unsupported>Here</big></a></strong></a>',
```

Reply:
fixed in 639a01f324 - yes it's a typo - thanks!